### PR TITLE
add new nurse consult component

### PR DESCRIPTION
**Pull request by mogoodrich**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java [hunk 1]
```diff
@@ -225,6 +225,10 @@ private void loadAppsAndExtensions() throws UnsupportedEncodingException {
             enableConsultInitial();
         }
 
+		if (config.isComponentEnabled(Components.NURSE_CONSULT)) {
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java:228]: If the "nurse consult" component is enabled, enable the nurse consult..

File: api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java [hunk 2]
```diff
@@ -656,6 +660,17 @@ private void enableConsultInitial() {
                 sessionLocationHasTag("Consult Note Location")));
     }
 
+	private void enableNurseConsult() {
+		extensions.add(visitAction(Extensions.NURSE_CONSULT_NOTE_VISIT_ACTION,
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java:664]: add to the list of "visit actions" a "nurse consult" visit action, which opens the form "nurseConsult.xml".  use the stethoscope icon; only display if the current location is tagged as a "Consult Note Location"; available to all users, no specific privileges required (we likely want to tweak this)

File: api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderConstants.java [hunk 3]
```diff
@@ -134,6 +134,7 @@
         public static final String VITALS_CAPTURE_VISIT_ACTION = "pih.form.vitals";
         public static final String CONSULT_NOTE_VISIT_ACTION = "pih.form.consult";
         public static final String CONSULT_NOTE_INITIAL_VISIT_ACTION = "pih.form.consultInitial";
+		public static final String NURSE_CONSULT_NOTE_VISIT_ACTION = "pih.form.nurseConsult";
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderConstants.java:137]: static constant used to name the new "consult initial" button we are adding under visit actions

File: api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderConstants.java [hunk 4]
```diff
@@ -382,6 +383,7 @@
             Extensions.PRIMARY_CARE_ADULT_FOLLOWUP_VISIT_ACTION,
             Extensions.PRIMARY_CARE_PEDS_INITIAL_VISIT_ACTION,
             Extensions.PRIMARY_CARE_PEDS_FOLLOWUP_VISIT_ACTION,
+			Extensions.NURSE_CONSULT_NOTE_VISIT_ACTION,
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderConstants.java:386]: This list lets us sort where the nurse consult action would appear amongst the list of visit actions

**Review by MiguelAHPpih — APPROVED** [review]

**Pull request merged**